sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2045,6 +2047,21 @@ object EliminateUnions extends Rule[LogicalPlan] {
 }
 
 /**
+ * Converts foldable numeric expressions to integers in [[GlobalLimit]] and [[LocalLimit]] operators
+ */
+object ResolveLimits extends Rule[LogicalPlan] {
```

Comment:
We can create a JIRA first and discuss with others to decide if it's useful to support all integral types in limit.